lang/java/avro/src/main/java/org/apache/avro/file/CodecFactory.java

```diff
@@ -69,11 +69,28 @@ public static CodecFactory bzip2Codec() {
     return new BZip2Codec.Option();
   }
 
-  /** zstandard codec.*/
-  public static CodecFactory zstandardCodec() {
-    return new ZstandardCodec.Option();
+  /** zstandard codec, with specific compression level.
+   *
+   * @param level The compression level should be between
+   * -5 and 22, inclusive.  Negative levels are 'fast' modes akin to
+   * lz4 or snappy, levels above 9 are generally for archival purposes,
+   * and levels above 18 use a lot of memory.*/
+  public static CodecFactory zstandardCodec(int level) {
+    return new ZstandardCodec.Option(level, false);
   }
 
+  /** zstandard codec, with specific compression level.
+  *
+  * @param level The compression level should be between
+  * -5 and 22, inclusive.  Negative levels are 'fast' modes akin to
+  * lz4 or snappy, levels above 9 are generally for archival purposes,
+  * and levels above 18 use a lot of memory.
+  * @param useChecksum if true, will include a checksum with each data block*/
+ public static CodecFactory zstandardCodec(int level, boolean useChecksum) {
+   return new ZstandardCodec.Option(level, false);
+ }
+
```

Comment:
I believe the useChecksum should be passed in here?

Reply:
Yup.  Fix inbound.